src/main/scala/org/broadinstitute/dsde/workbench/leonardo/dns/ClusterDnsCache.scala

```diff
@@ -51,41 +40,29 @@ object ClusterDnsCache {
   * @param proxyConfig the proxy configuration
   * @param dbRef provides access to the database
   */
-class ClusterDnsCache(proxyConfig: ProxyConfig, dbRef: DbReference) extends Actor with LazyLogging {
-  var ProjectNameToHost: Map[(GoogleProject, ClusterName), GetClusterResponse] = Map.empty
-
-  import context.dispatcher
-
-  override def preStart(): Unit = {
-    super.preStart()
-    self ! RefreshFromDatabase
-  }
-
-  override def receive: Receive = {
-    case RefreshFromDatabase =>
-      queryForClusters pipeTo self
-
-    case ProcessClusters(clusters) =>
-      processClusters(clusters)
-      scheduleRefresh
-
-    // add one ready cluster to the caches now instead of waiting for the refresh cycle
-    case ProcessReadyCluster(cluster) =>
-      sender ! processReadyCluster(cluster)
-
-    case GetByProjectAndName(googleProject, clusterName) =>
-      sender ! ProjectNameToHost.getOrElse((googleProject, clusterName), ClusterNotFound)
-  }
-
-  private def scheduleRefresh = {
-    context.system.scheduler.scheduleOnce(proxyConfig.dnsPollPeriod, self, RefreshFromDatabase)
-  }
-
-  private def queryForClusters: Future[ProcessClusters] ={
-    dbRef.inTransaction { dataAccess =>
-      dataAccess.clusterQuery.listActive()
-    }.map(ProcessClusters.apply)
-  }
+class ClusterDnsCache(proxyConfig: ProxyConfig, dbRef: DbReference)(implicit  executionContext: ExecutionContext) extends LazyLogging {
+
+  val ProjectNameToHost = CacheBuilder.newBuilder()
+    .expireAfterWrite(5, TimeUnit.SECONDS)
```

Comment:
Agreed, I came up with that number considering  that when a notebook is opened most of the call would happen within that time period or less. It would also prevent the cache from having stale data. I was planning on using load test results to tweak that duration.